modules/auxiliary/scanner/dns/dyn_dns_update.rb

```diff
@@ -0,0 +1,178 @@
+# -*- coding: binary -*-
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+require 'msf/core'
+require 'net/dns'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::Udp
+
+  def initialize
+    super(
+        'Name'           => 'DNS Server Dynamic Update Record Injection',
+        'Description'    => %q{
+        This module allows adding and/or deleting a record to
+        any remote DNS server that allows unrestricted dynamic updates.},
+        'Author'         => [ 'King Sabri <king.sabri[at]gmail.com>' ],
+        'References'     =>
+            [
+                ['URL', 'http://www.tenable.com/plugins/index.php?view=single&id=35372'],
+                ['URL', 'https://github.com/KINGSABRI/CVE-in-Ruby/tree/master/NONE-CVE/DNSInject'],
+                ['URL', 'https://github.com/ChrisTruncer/PenTestScripts/blob/master/DNSInject.py']
+            ],
+        'License'        => MSF_LICENSE,
+        'Actions'        =>
+            [
+                ['ADD',  {'Description' => 'Add a new record. [Default]'}],
+                ['DEL',  {'Description' => 'Delete an existing record.'}]
+            ],
+        'DefaultAction' => 'ADD'
+    )
+
+    register_options(
+        [
+            OptString.new('DOMAIN', [true, 'The domain name']),
+            OptAddress.new('NS', [true, 'The vulnerable DNS server IP address']),
+            OptString.new('INJECTDOMAIN', [true, 'The name record you want to inject']),
+            OptAddress.new('INJECTIP', [true, 'The IP you want to assign to the injected record']),
+        ], self.class)
+
+    register_advanced_options(
+        [
+            Opt::RPORT(53),
+            OptInt.new('TIMEOUT', [false, 'DNS TIMEOUT', 8]),
+            OptInt.new('RETRY', [false, 'Number of times to try to resolve a record if no response is received', 2]),
+            OptInt.new('RETRY_INTERVAL', [false, 'Number of seconds to wait before doing a retry', 2]),
+            OptBool.new('TCP_DNS', [false, 'Run queries over TCP', false])
+        ], self.class)
+
+    deregister_options( 'RHOST', 'RPORT' )
+
+  end
+
+  #
+  # Update String class for easir domain and IP processing
+  #
+  class ::String
```

Comment:
Yes, I see, but I couldn't find an elegant way to do it and keep user outputs as it should be. 

pls advise